cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -272,4 +248,90 @@ private String findBashOnWindows() {
     throw new IllegalStateException("Could not find Bash. Please install Git for Windows and rerun.");
   }
 
+  private void handleCapture(Process process, List<String> out, List<String> err) throws IOException {
+
+    try (BufferedReader outReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        BufferedReader errReader = new BufferedReader(new InputStreamReader(process.getErrorStream()))) {
+      String outLine = "";
+      String errLine = "";
+      while ((outLine != null) || (errLine != null)) {
+        if (outLine != null) {
+          outLine = outReader.readLine();
+          if (outLine != null) {
+            out.add(outLine);
+          }
+        }
+        if (errLine != null) {
+          errLine = errReader.readLine();
+          if (errLine != null) {
+            err.add(errLine);
+          }
+        }
+      }
+    }
+  }
+
+  private boolean checkAndHandlePossibleBashScript(String executableName) {
+
+    String fileExtension = FilenameUtil.getExtension(executableName);
+    boolean isBashScript = "sh".equals(fileExtension) || hasSheBang(this.executable);
+    if (isBashScript) {
+      String bash = "bash";
+      if (this.context.getSystemInfo().isWindows()) {
+        String findBashOnWindowsResult = findBashOnWindows();
+        if (findBashOnWindowsResult != null) {
+          bash = findBashOnWindowsResult;
+        }
+      }
+      this.arguments.add(0, bash);
+    }
+    return isBashScript;
+  }
+
+  private void performLogOnError(ProcessResult result, int exitCode) {
+
+    if (!result.isSuccessful() && (this.errorHandling != ProcessErrorHandling.NONE)) {
+      String message = createCommandMessage(" failed with exit code " + exitCode + "!");
+      if (this.errorHandling == ProcessErrorHandling.THROW) {
+        throw new CliException(message, exitCode);
+      }
+      IdeSubLogger level;
+      if (this.errorHandling == ProcessErrorHandling.ERROR) {
+        level = this.context.error();
+      } else if (this.errorHandling == ProcessErrorHandling.WARNING) {
+        level = this.context.warning();
+      } else {
+        level = this.context.error();
+        level.log("Internal error: Undefined error handling {}", this.errorHandling);
+      }
+      level.log(message);
+    }
+  }
+
+  private void modifyArgumentsOnBackgroundProcess(boolean isBashScript) {
+
+    if (context.getSystemInfo().isWindows() && !isBashScript) {
+
+      this.context.warning(
+          "Currently starting the process as background process in windows will use 'more' command to redirect output to a new cmd window!");
+
+      this.processBuilder.redirectOutput(Redirect.PIPE).redirectError(Redirect.PIPE);
+      List<String> windowsArgs = List.of("cmd.exe", "/c", "start", "cmd.exe", "/k");
+
+      String currentCommandToRunInCmd = this.arguments.stream().map(Object::toString).collect(Collectors.joining(" "));
+      currentCommandToRunInCmd += " | more";
```

Comment:
https://www.baeldung.com/linux/detach-process-from-terminal

Reply:
Or as an alternative we can run the process without getting the output. In the simplest way we simply then add `> /dev/null`.

Reply:
So, that means I should assume that all Windows users have a bash and discard the output as "fallback" if no bash installation can be found?

Reply:
It was always a per-requisit of devonfw-ide that the user has git installed. This is still the same for IDEasy since we are calling git to clone the settings and the urls as external process. On Windows if you install git, you always have git-bash as well. On Linux and MacOS every user has bash even in a vanilla installation (Except some exotic linux distributions for pocket calculators or C64). So having bash is a given on any supported OS.

Whether the output is suppressed or rushes through the shell of the user should be parameterized. Ideally we introduce an enum state for this:
* suppress output (`>/dev/null 2>&1`)
* feed output into current terminal (run `bash -c` and pass the command with an added `&` like for the option above)
* open a new window with the output (could be done in the way it is implemented currently with CMD on windows but on linux/mac we need to figure out how to do that as we need to find the according terminal app and on MacOS there is not only the default MacOS Terminal but als iTerm2). Feel free to leave this option out for the moment but I can see this quite valuable for the future as well. And maybe we will allow some user specific configuration to configure one of these 3 options as default and per individual tool in the future... Let us always thing further and beyond but also continue with small agile steps from one working state to the next...

Reply:
BTW: If we combine all this in an enum, we could also replace the `capture` and `background` flags what prevents inconsistent combinations like `background=true & capture=true` and also increases readability of code:
~~pc.run(false, true, true);~~
```
pc.run(ProcessMode.BACKGROUND_SILENT);
```

Reply:
I've implemented a ProcessMode enum (Thx for the suggestion, that makes the implenetation way easier to handle than having all possibilities as boolean parameters...) and refactored the code. After some local testing in Linux and Windows bash I think our desired behavior can only be implemented using the unix "disown" (See explanation in added JavaDoc). Instead of redirecting to the null file using '>/dev/null 2>&1', I've just used the  ProcessBuilder.Redirec.DISCARD which does the same OS independently. I've removed the open new window possibilty for now and added a todo comment in the ProcessMode enum.

Reply:
Excellent - you solved this perfectly. Thanks 👍